conbench/entities/commit.py

```diff
@@ -413,9 +441,27 @@ def _parse_commit(commit):
             "author_avatar": author["avatar_url"] if author else None,
         }
 
-    def _get_response(self, url):
+    def _get_response(self, url) -> dict:
+        """
+        Return deserialized JSON-structure or raise an exception.
+        """
+        # This can raise exceptions corresponding to transient issues related
+        # to DNS, TCP, HTTP during sending request, while waiting for response,
+        # or while receiving the response.
         response = self.session.get(url) if self.session else requests.get(url)
+
+        # An HTTP response was received.
         if response.status_code != 200:
-            print(response.json())
-            return None
+            # This is here to log details an make debugging easier.
+            log.info(
+                "got unexpected HTTP response with code %s: %s",
+                response.status_code,
+                response.text,
+            )
+
+        # Raise an exception for all known HTTP error response types.
+        response.raise_for_status()
+
+        # This may raise an exception if JSON-deserialization fails. If JSON
+        # deser succeeds then this is known to be a dict at the outest level.
```

Comment:
Might raise `ValueError` if JSON-deserialization fails, i.e. that particular exception is not from the `requests.exceptions` hierarchy. Have not tested though.

Reply:
Yeah, or in particular if the response has no content I think `.json()` will fail. But I don't think any of the endpoints we hit have that behavior.

Reply:
Yes, right.

To summarize: once we're here and JSON deserialization fails this means we got a 2xx response from the GitHub HTTP/JSON API _without_ JSON body in the response. Which _should not_ happen. But since it's not in control of this program here, it's part of the possible outcomes of this function, in the strictest sense.